From 2534857e7e06b3c6ba1bf5824db24d8e73ea677e Mon Sep 17 00:00:00 2001
From: David Milburn <dmilburn@redhat.com>
Date: Tue, 13 Jun 2017 16:36:50 -0500
Subject: nvme: submit nvme_admin_activate_fw to admin queue
References: bsc#1044532
Patch-mainline: Never, SLES12 specific

The nvme_admin_activate_fw command should be submitted to
the admin queue; otherwise, mode 0x0F (ACTIVATE_DEFERRED_MICROCODE)
and mode 0x05 (DOWNLOAD_SAVE_ACTIVATE) will fail.

Fixes: b90c48d0c11ef("nvme: split nvme_trans_send_fw_cmd")
Reported-by: Joseph Szczypek <joseph.szczypek@hpe.com>
Tested-by: Micah Parrish <micah.parrish@hpe.com>
Signed-off-by: David Milburn <dmilburn@redhat.com>
Signed-off-by: Johannes Thumshirn <jthumshirn@suse.de>
---
 drivers/nvme/host/scsi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/nvme/host/scsi.c b/drivers/nvme/host/scsi.c
index e947e298a737..836f374ce0bb 100644
--- a/drivers/nvme/host/scsi.c
+++ b/drivers/nvme/host/scsi.c
@@ -1297,7 +1297,7 @@ static int nvme_trans_send_activate_fw_cmd(struct nvme_ns *ns, struct sg_io_hdr
 	c.common.opcode = nvme_admin_activate_fw;
 	c.common.cdw10[0] = cpu_to_le32(buffer_id | NVME_FWACT_REPL_ACTV);
 
-	nvme_sc = nvme_submit_sync_cmd(ns->queue, &c, NULL, 0);
+	nvme_sc = nvme_submit_sync_cmd(ns->ctrl->admin_q, &c, NULL, 0);
 	return nvme_trans_status_code(hdr, nvme_sc);
 }
 
-- 
2.12.3

